langchain4j-agentic/src/main/java/dev/langchain4j/agentic/internal/AbstractService.java

```diff
@@ -0,0 +1,63 @@
+package dev.langchain4j.agentic.internal;
+
+import dev.langchain4j.agentic.cognisphere.Cognisphere;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static dev.langchain4j.agentic.internal.AgentUtil.agentsToExecutors;
+
+public abstract class AbstractService<T, S> {
+
+    private static final Function<Cognisphere, Object> DEFAULT_OUTPUT_FUNCTION = cognisphere -> null;
+    private static final Consumer<Cognisphere> DEFAULT_INIT_FUNCTION = cognisphere -> { };
+
+    protected final Class<T> agentServiceClass;
+
+    protected Consumer<Cognisphere> beforeCall = DEFAULT_INIT_FUNCTION;
+
+    protected String outputName;
+    protected Function<Cognisphere, Object> output = DEFAULT_OUTPUT_FUNCTION;
```

Comment:
```suggestion
    protected Function<Cognisphere, CompletionStage<Object>> output = DEFAULT_OUTPUT_FUNCTION;
```
Shouldn't it be asynchronous from day one?

Reply:
This function doesn't (normally) imply any I/O, it is just a transformation eventually applied to the output produced by an agent or set of agents before storing the result in the `Cognisphere` like in this example https://github.com/langchain4j/langchain4j/pull/3458/files#diff-c9bf8b215cb35eda3e835bdd831ee2e62e4fbd8aaf5617b1e8d8462d238e92e2R358